services/api/src/unified_graphics/diag.py

```diff
@@ -81,6 +96,26 @@ def open_diagnostic(variable: Variable, loop: MinimLoop) -> xr.Dataset:
     return xr.open_dataset(diag_file)
 
 
+def open_s3_diagnostic(diag_uri: str, filename: str) -> xr.Dataset:
+    """Opens a diag file in S3. Assumes AWS S3, and grabs credentials via Boto3 defaults."""
+
+    bucket = s3fs.S3FileSystem(anon=False)
+    diag_file = diag_uri + filename  # Path doesn't support file URIs
+
+    # xarray.open_dataset doesn't distinguish between a file it can't understand
+    # and a file that's not there. It raises a ValueError even for missing
+    # files. We raise a FileNotFoundError to make debugging easier.
+    if not bucket.exists(diag_file):
+        print(f"No such file: '{str(diag_file)}'")
```

Comment:
I’m pretty sure we have logging configured, so this would be better as

```suggestion
        current_app.logger.error(f"No such file: '{str(diag_file)}'")
```

Reply:
Oh, that's a great catch. Thanks! I'll go ahead and switch those to logging statements.